src/main/java/org/kairosdb/core/http/rest/QueryAutocompleter.java

```diff
@@ -0,0 +1,49 @@
+package org.kairosdb.core.http.rest;
+
+import com.google.common.collect.SetMultimap;
+import org.kairosdb.core.datastore.QueryMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class QueryAutocompleter {
+    private static final Logger logger = LoggerFactory.getLogger(QueryAutocompleter.class);
+
+    public void complete(QueryMetric query) {
+        completeMetricTag(query);
+    }
+
+    private void completeMetricTag(QueryMetric query) {
```

Comment:
Two points which should be included in the implementation:
1. We should not add more `metric` tags if the query already contains a `metric` tag.
2. I think we should not do any of the auto-completion if the `key` is not wildcard
I don't see the above 2 points implemented. @alexkorotkikh What do you think?

Reply:
1. Good catch, thanks! I've added this check to implementation
2. I think we decided to add metric even if key is not a wildcard. It makes an implementation a bit more straightforward because of fewer conditions, while not changing its semantic (metric tag has lower priority than key tag in split index selection, and will be ignored if key is not a wildcard)